optimum/exporters/onnx/base.py

```diff
@@ -473,13 +474,80 @@ def flatten_output_collection_property(self, name: str, field: Iterable[Any]) ->
         return flattened_output
 
 
+class ConfigBehavior(str, enum.Enum):
+    """
+    Specifies the behavior of the [`~exporters.onnx.base.OnnxSeq2SeqConfigWithPast`]:
+        - MONOLITH: the config can be used to export the whole seq2seq model as a single file.
+        - ENCODER: the config can be used to export the encoder part of the seq2seq model.
+        - DECODER: the config can be used to export the decoder part of the seq2seq model.
+    """
+
+    MONOLITH = "monolith"
+    ENCODER = "encoder"
+    DECODER = "decoder"
+
+
 class OnnxSeq2SeqConfigWithPast(OnnxConfigWithPast):
     """
     Inherits from [`~exporters.onnx.OnnxConfigWithPast`]. A base class to handle the ONNX configuration of encoder-decoder models.
     """
 
     PAD_ATTENTION_MASK_TO_MATCH_TOTAL_SEQUENCE_LENGTH = False
 
+    def __init__(
+        self,
+        config: "PretrainedConfig",
+        task: str = "default",
+        patching_specs: Optional[List[PatchingSpec]] = None,
+        use_past: bool = False,
+        use_past_in_inputs: Optional[bool] = None,
+        use_present_in_outputs: Optional[bool] = None,
+        behavior: ConfigBehavior = ConfigBehavior.MONOLITH,
+    ):
+        super().__init__(
+            config,
+            task=task,
+            patching_specs=patching_specs,
+            use_past=use_past,
+            use_past_in_inputs=use_past_in_inputs,
+            use_present_in_outputs=use_present_in_outputs,
+        )
+        self._behavior = behavior
+        self.override_attributes_for_behavior()
+
+    def override_attributes_for_behavior(self):
+        """Override this to specify custom attribute change for a given behavior."""
+        if self._behavior is ConfigBehavior.ENCODER:
+            self.task = "default"
+        if self._behavior is ConfigBehavior.DECODER:
+            self.use_past_in_inputs = self.use_past
+            self.use_present_in_outputs = True
+
+    def with_behavior(
+        self, behavior: Union[str, ConfigBehavior], use_past: bool = False
+    ) -> "OnnxSeq2SeqConfigWithPast":
+        """
+        Creates a copy of the current OnnxConfig but with a different `ConfigBehavior` and `use_past` value.
+
+        Args:
+            behavior ([`ConfigBehavior`]):
+                The behavior to use for the new instance.
+            use_past (`bool`, defaults to `False`):
+                Whether or not the new instance should use past.
+
+        Returns:
+            `OnnxSeq2SeqConfigWithPast`
+        """
+        if isinstance(behavior, str) and not isinstance(behavior, ConfigBehavior):
+            behavior = ConfigBehavior(behavior)
```

Comment:
So `ConfigBehavior` is an `Enum` with string values. Doing this will check as you mention. But it also allows to do something like: 

```python
if self._behavior is ConfigBehavior.DECODER
```

Because if `self._behavior` is a string, you need to check for equality instead, with something like:

```python
if self._behavior.value == "decoder"
```

Reply:
Yes alright. But since `ConfigBehavior` inherits from `str` the check `ConfigBehavior.MONOLITH == "monolith"` goes just fine.

Reply:
Yes, but not with the `is` operator. Which I think is nice in terms of readability (and is much faster too)

Reply:
yes yes agreed, just wanted to understand the motive :)